### PR TITLE
split AllMethods, AllFields and AllConstructors

### PR DESCRIPTION
**Pull request by SergeyLabutin**

### CODE REVIEW

File: jtransc-core/src/com/jtransc/plugin/reflection/MetaReflectionJTranscPlugin.kt [hunk 1]
```diff
@@ -487,17 +487,18 @@ class MetaReflectionJTranscPlugin : JTranscPlugin() {
...
+
 				val additionalMethods: MutableList<AstMethod> = mutableListOf()
 				var methodIndex: Int = 0
 				while (methodIndex * CASES_PER_SWITCH < list.size) {
 					val mI: Int = methodIndex
 					val sI: Int = methodIndex * CASES_PER_SWITCH
-					val parentClass: AstClass = clazz
-					val parentMethod: AstMethod = method
+					val parentClass: AstClass = classes[methodIndex % classes.size]
+					val parentMethod: AstMethod? = parentClass.getMethodWithoutOverrides(methodName)
```
Comment (soywiz) [inline jtransc-core/src/com/jtransc/plugin/reflection/MetaReflectionJTranscPlugin.kt]: Have you tested this? Maybe we should add a test to big test with CASES_PER_SWITCH * 3 or things like that and tons of methods and fields and check reflection. The problem is that tests may pass with a few classes or small classes and production code with tons of classes fail, or not work as expected with some edge cases

Comment (soywiz) [inline jtransc-core/src/com/jtransc/plugin/reflection/MetaReflectionJTranscPlugin.kt]: As far as I remember `getMethodWithoutOverrides` return null if the method has overrides. So you can't match multiple methods with the same name with this I think

**Comment by SergeyLabutin** [comment]
I not change logic, split to 3 classes only, test on my project and test. I really not understand problem =) All 3 classes keeps for replace body, each class have only one function.
in my case i have 3 classes with 20 000 lines in haxe and small that 50 000 in cpp. Before we have one class with 60 000 lines in haxe and 150 000 lines in cpp. It is not use CASES_PER_SWITCH. Problem with big cases different. Yes both depends of project. I not create universal, but if anybody have big than my project, we can write more classes.

P.S. Maybe i not understand you =(

**Comment by soywiz** [comment]
You are right. Nervermind
